spring-cloud-dataflow-module-deployers/spring-cloud-dataflow-module-deployer-cloudfoundry/src/main/java/org/springframework/cloud/dataflow/module/deployer/cloudfoundry/ModuleStatusBuilder.java

```diff
@@ -56,9 +56,15 @@ ModuleStatusBuilder withId(ModuleDeploymentId id) {
 		return this;
 	}
 
-	// TODO: Find out what the application state candidates are
 	private static ModuleStatus.State convertState(String state) {
 		switch (state) {
+			case "STARTED":
+			case "STARTING":
+			case "STOPPED":
```

Comment:
Good point on the caching; that will be nice when we're taking advantage of it. In any case, this is a topic to explore another day. For now STOPPED -> "deploying" makes perfect sense. Thanks for the detail.